File: src/main/java/xyz/realmath/ArrayTuples.java
package xyz.realmath;

public final class ArrayTuples {
  public static ArrayTuple<Rational> rationalTuple(int... ints) {
    Rational[] rationals = new Rational[ints.length];
    for (int i = 0; i < ints.length; i++) {
      rationals[i] = ints[i];
    }
    return new ArrayTuple<>(rationals);
  }

  public static ArrayTuple<MultiVarPolynomial> polynomialTuple(int... ints) {
    MultiVarPolynomial[] pols = new MultiVarPolynomial[ints.length];
    for (int i = 0; i < ints.length; i++) {
      pols[i] = ints[i];
    }
    return new ArrayTuple<>(pols);
  }
}

File: src/main/kotlin/xyz/realmath/ArrayTuples.kt
package xyz.realmath

object ArrayTuples {
    fun rationalTuple(vararg ints: Int): ArrayTuple<Rational> {
        val rationals = Array(ints.size, { i -> Rational(ints[i]) })
        return ArrayTuple(*rationals)
    }

    fun polynomialTuple(vararg ints: Int): ArrayTuple<MultiVarPolynomial> {
        val pols = Array(ints.size, { i -> MultiVarPolynomial.valueOf(ints[i]) })
        return ArrayTuple(*pols)
    }
}
